api/restHandler/ExternalCiRestHandler.go

```diff
@@ -89,3 +106,61 @@ func (impl ExternalCiRestHandlerImpl) HandleExternalCiWebhook(w http.ResponseWri
 
 	common.WriteJsonResp(w, err, nil, http.StatusOK)
 }
+
+func (impl ExternalCiRestHandlerImpl) HandleExternalCiWebhookByApiToken(w http.ResponseWriter, r *http.Request) {
+	setupResponse(&w, r)
+	vars := mux.Vars(r)
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized", http.StatusUnauthorized)
+		return
+	}
+	externalCiId, err := strconv.Atoi(vars["externalCiId"])
+	if err != nil {
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	decoder := json.NewDecoder(r.Body)
+	var req pubsub.CiCompleteEvent
+	err = decoder.Decode(&req)
+	if err != nil {
+		impl.logger.Errorw("request err, HandleExternalCiWebhookByApiToken", "err", err, "payload", req)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	req.TriggeredBy = userId
+	impl.logger.Infow("request payload, HandleExternalCiWebhookByApiToken", "payload", req)
+
+	ciArtifactReq, err := impl.ciEventHandler.BuildCiArtifactRequestForWebhook(req)
```

Comment:
too much logic in resthandler, it should be pushed down to service also name of functions need change

Reply:
refactor done, only services calls and validation and auth in this rest handler.